### DISCUSSION TITLE
API workflow_dispatch not running in workflow

### DISCUSSION THREAD

**Discussion by arendondiosa** [community/community#25685] [Actions]
<p>Hi :+1:<br>
I’m using the GitHub actions API to run a workflow on a repository:</p>
<pre><code class="lang-auto">url = "{0}/repos/{1}/actions/workflows/{2}/dispatches".format(
                self.api_url, self.repository_path, workflow_id
            )
            headers = {
                "Authorization": "token %s" % self.token,
                "Accept": "application/vnd.github.v3+json",
            }
            data = {"ref": ref}

            response = requests.post(url, headers=headers, data=json.dumps(data))
</code></pre>
<p>When I run the request in the local environment, it works and I can see the flow running in the ACTIONS tab of the repository. When I run the same script in a job with GitHub Actions, the output is the same but the workflow doesn’t run (No errors - code 204)</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/4/849332cccfc96cda7a3afd0a51ef3763af496c29.png" data-download-href="/uploads/short-url/iUOoDd7HCxbmMjU5gxRgXQoE4Q1.png?dl=1" title="Screen Shot 2021-07-17 at 10.03.56" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/14989202/181092498-803669d4-4d74-4994-b6fb-f68087c2be07.png" alt="Screen Shot 2021-07-17 at 10.03.56" data-base62-sha1="iUOoDd7HCxbmMjU5gxRgXQoE4Q1" width="690" height="330" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/4/849332cccfc96cda7a3afd0a51ef3763af496c29_2_690x330.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/4/849332cccfc96cda7a3afd0a51ef3763af496c29_2_1035x495.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/8/4/849332cccfc96cda7a3afd0a51ef3763af496c29.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/4/849332cccfc96cda7a3afd0a51ef3763af496c29_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2021-07-17 at 10.03.56</span><span class="informations">1150×550 67.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Any ideas? :confused:</p>
<p>PS: I’m using the same configuration in both tests (Local env and GitHub)</p>

**Comment by shrink** [comment, marked as answer]
<p>The default token available to a GitHub Action (<code>GITHUB_TOKEN</code>) cannot trigger other Actions.</p>
<blockquote>
<p>When you use the repository’s <code>GITHUB_TOKEN</code> to perform tasks on behalf of the GitHub Actions app, events triggered by the <code>GITHUB_TOKEN</code> will not create a new workflow run<br>
<sub>– <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow" rel="noopener nofollow ugc">Using the <code>GITHUB_TOKEN</code> in a workflow</a></sub></p>
</blockquote>
<p>The reason this works locally is that you’re authenticated as yourself, so for this to work inside a GitHub Action you’ll need to use a <a href="https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token" rel="noopener nofollow ugc">Personal Access Token</a> instead.</p>

**Comment by arendondiosa** [comment]
<p>Thanks <a class="mention" href="/shrink">@shrink</a>, you’re right :muscle: . I was using <code>github.token</code> instead of <code>secrets.ACCESS_TOKEN</code>.</p>
<p>I was confused because the API endpoint response did not have an error.</p>
